webhook.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2021 The Cloud Native Events Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"sync"
+
+	"github.com/redhat-cne/sdk-go/pkg/util/wait"
+
+	"github.com/gorilla/mux"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+
+	"net/http"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var singleton sync.Once
+
+// WebhookInstance ... is singleton instance
+var WebhookInstance *Webhook
+
+// Webhook defines rest routes server object
+type Webhook struct {
+	port       int
+	apiPath    string
+	closeCh    <-chan struct{}
+	HTTPClient *http.Client
+	httpServer *http.Server
+	status     serverStatus
+	targetPub  *pubsub.PubSub
+}
+
+// HwEvent ... temporary format. TODO: define schema for redfish hw event data
+type HwEvent struct {
+	ID   string `json:"Id"`
+	Name string `json:"Name"`
+}
+
+// InitWebhook is used to supply configurations for rest routes server
+func InitWebhook(port int, apiPath string, targetPub *pubsub.PubSub) *Webhook {
+	singleton.Do(func() {
+		WebhookInstance = &Webhook{
+			port:    port,
+			apiPath: apiPath,
+			status:  notReady,
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					MaxIdleConnsPerHost: 20,
+				},
+				Timeout: 10 * time.Second,
+			},
+			targetPub: targetPub,
+		}
+	})
+	// singleton
+	return WebhookInstance
+}
+
+// EndPointHealthChk checks for rest service health
+func (s *Webhook) EndPointHealthChk() (err error) {
+	log.Info("checking for rest service health\n")
```

Comment:
Do don't need this , it was required for pubsub rest API only  because its default in side car